stem/solver.py

```diff
@@ -1,10 +1,488 @@
+        """
+        return "newton_raphson"
+
+
+@dataclass
+class LineSearchStrategy(StrategyTypeABC):
+    """
+    Class containing information about the line search strategy
+
+    Inheritance:
+        - :class:`StrategyTypeABC`
+
+    Attributes:
+        - max_line_search_iterations (int): maximum number of line search iterations. Default value is 10.
+        - first_alpha_value (float): first alpha guess value used for the first iteration. Default value is 1.0.
+        - second_alpha_value (float): second alpha guess value used for the first iteration. Default value is 0.5.
+        - min_alpha (float): minimum possible alpha value at the end of the algorithm. Default value is 1e-4.
+        - max_alpha (float): maximum possible alpha value at the end of the algorithm. Default value is 1e4.
+        - line_search_tolerance (float): Tolerance of the line search algorithm, defined as the ratio between maximum\
+            residual*alpha*dx and current iteration residual*alpha*dx. Default value is 1e-4.
+        - echo_level (int): echo level. Default value is 0.
+    """
+    max_line_search_iterations: int = 10
+    first_alpha_value: float = 1.0
+    second_alpha_value: float = 0.5
+    min_alpha: float = 1e-4
+    max_alpha: float = 1e4
+    line_search_tolerance: float = 1e-4
+    echo_level: int = 0
+
+    @property
+    def strategy_type(self):
+        """
+        Returns the strategy type name of the line search strategy
+
+        Returns:
+            - str: strategy type name
+        """
+        return "line_search"
+
+
+@dataclass
+class ArcLengthStrategy(StrategyTypeABC):
+    """
+    Class containing information about the arc length strategy
+
+    Inheritance:
+        - :class:`StrategyTypeABC`
+
+    Attributes:
+        - desired_iterations (int): This is used to calculate the radius of the next step. Default value is 10.
+        - max_radius_factor (float): maximum radius factor of the arc. Default value is 1.0.
+        - min_radius_factor (float): minimum radius factor of the arc. Default value is 0.1.
+
+    """
+    desired_iterations: int = 10
+    max_radius_factor: float = 1.0
+    min_radius_factor: float = 0.1
+
+    @property
+    def strategy_type(self):
+        """
+        Returns the strategy type name of the arc length strategy
+
+        Returns:
+            - str: strategy type name
+
+        """
+        return "arc_length"
+
+
+@dataclass
+class LinearSolverSettingsABC(ABC):
+    """
+    Class containing information about the linear solver settings
+
+    Attributes:
+        - scaling (bool): if true, the system matrix will be scaled before solving the linear system of equations.\
+            Default value is False.
+
+    """
+    scaling: bool = False
+
+    @property
+    @abc.abstractmethod
+    def solver_type(self):
+        raise Exception("abstract class of linear solver settings is called")
+
+
+@dataclass
+class Amgcl(LinearSolverSettingsABC):
```

Comment:
there is only one Linear solver?

Reply:
more are available in Kratos, but for me amgcl works well. I'd say, lets keep the option open to add more in the future